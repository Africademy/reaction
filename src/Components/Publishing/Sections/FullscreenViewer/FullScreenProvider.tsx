import { isFunction } from "lodash"
import PropTypes from "prop-types"
import { Component } from "react"

interface Props {
  children: any
}

interface State {
  viewerIsOpen: boolean
  slideIndex: number
}

export class FullScreenProvider extends Component<Props, State> {
  state = {
    slideIndex: 0,
    viewerIsOpen: false,
  }

  static childContextTypes = {
    onViewFullscreen: PropTypes.func.isRequired,
    openViewer: PropTypes.func.isRequired,
    closeViewer: PropTypes.func.isRequired,
    slideIndex: PropTypes.number.isRequired,
    viewerIsOpen: PropTypes.bool.isRequired,
  }

  getChildContext() {
    return {
      onViewFullscreen: this.openViewer,
      openViewer: this.openViewer,
      closeViewer: this.closeViewer,
      slideIndex: this.state.slideIndex,
      viewerIsOpen: this.state.viewerIsOpen,
    }
  }

  openViewer = index => {
    const body = document.getElementsByTagName("BODY")[0]
    body.setAttribute("style", "overflow: hidden;")

    this.setState({
      viewerIsOpen: true,
      slideIndex: index,
    })
  }

  closeViewer = () => {
    const body = document.getElementsByTagName("BODY")[0]
    body.setAttribute("style", "overflow: scroll;")
    this.setState({
      viewerIsOpen: false,
    })
  }

  render() {
    const { slideIndex, viewerIsOpen } = this.state

    if (isFunction(this.props.children)) {
      return this.props.children({
        slideIndex,
        viewerIsOpen,
        openViewer: this.openViewer,
        closeViewer: this.closeViewer,
      })
    } else {
      return this.props.children
    }
  }
}
