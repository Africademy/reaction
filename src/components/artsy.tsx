import * as PropTypes from "prop-types"
import * as React from "react"
import * as sharify from "sharify"

/**
 * The currently signed-in user.
 */
export interface CurrentUser extends sharify.CurrentUser {}

/**
 * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextConsumer}
 *
 * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands weâ€™ll be able to not make this
 *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
 */
export interface ContextProps {
  artsy?: Props
}

interface Props {
  currentUser?: CurrentUser,
}

const ContextTypes = {
  artsy: PropTypes.shape({
    currentUser: PropTypes.object,
  }),
}

/**
 * This component provides Artsy specific context props to components down the tree. Components that wish to consume
 * these props should be wrapped with the `ContextConsumer` higher-order component.
 *
 * @see {@link ContextConsumer}
 */
export class ContextProvider extends React.Component<Props, null> {
  static childContextTypes = ContextTypes

  constructor(props) {
    if (React.Children.count(props.children) > 1) {
      throw new Error("A ContextProvider expects a single child.")
    }
    super(props)
  }

  getChildContext(): ContextProps {
    return {
      artsy: {
        currentUser: this.props.currentUser,
      },
    }
  }

  render() {
    return this.props.children as any
  }
}

/**
 * Wraps a component in a higher-order component that injects the Artsy specific context props into the given component.
 * Wrapped components are expected to be rendered in a tree that contains a `ContextProvider` component.
 *
 * The injected props are described in `ContextProps`.
 *
 * @see {@link ContextProvider}
 * @see {@link ContextProps}
 *
 * @param Component The component that needs Artsy specific context props.
 */
export function ContextConsumer<P>(
  Component: React.ComponentClass<P> | React.StatelessComponent<P>,
): React.ComponentClass<P> {
  const name = Component.displayName || Component.name
  return class extends React.Component<P, void> {
    static contextTypes = ContextTypes
    static displayName = `Artsy(${name})`

    constructor(props, context) {
      if (!context.artsy) {
        const start = name || "A component"
        throw new Error(`${start}, which needs Artsy props, was not wrapped inside a ContextProvider component.`)
      }
      super(props, context)
    }

    render() {
      return <Component {...this.props} {...this.context} />
    }
  }
}
