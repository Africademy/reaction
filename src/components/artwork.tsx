import * as React from "react"
import * as Relay from "react-relay"
import styled from "styled-components"

const Image = styled.img`
  width: 100%;
`

export class Artwork extends React.Component<RelayProps, null> {
  render() {
    return (
      <div>
        <Image src={this.props.artwork.image.url} />
        <p>{this.props.artwork.title}</p>
      </div>
    )
  }
}

export default Relay.createContainer(Artwork, {
  fragments: {
    artwork: () => Relay.QL`
      fragment on Artwork {
        title
        date
        sale_message
        is_in_auction
        image {
          url(version: "large")
          aspect_ratio
        }
        artists {
          name
        }
        partner {
          name
        }
        href
      }
    `
  },
})

interface RelayProps {
  artwork: {
    title: string | null,
    date: string | null,
    sale_message: string | null,
    is_in_auction: boolean | null,
    image: {
      url: string | null,
      aspect_ratio: number | null,
    } | null,
    artists: Array<{
      name: string | null,
    } | null> | null,
    partner: {
      name: string | null,
    } | null,
    href: string | null,
  },
}
