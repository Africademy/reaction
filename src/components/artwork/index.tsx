import * as React from "react"
import * as Relay from "react-relay"
import styled from "styled-components"
import colors from "../../assets/colors"
import ArtworkMetadata from "./metadata"

const Image = styled.img`
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
`

const Placeholder = styled.div`
  background-color: ${colors.grayMedium};
  position: relative;
  width 100%;
`

interface Props extends RelayProps, React.HTMLProps<Artwork> {
  style?: any
}

export class Artwork extends React.Component<Props, null> {
  render() {
    return (
      <div style={this.props.style}>
        <Placeholder style={{ paddingBottom: this.props.artwork.image.placeholder }}>
          <Image src={this.props.artwork.image.url} />
        </Placeholder>
        <ArtworkMetadata artwork={this.props.artwork} />
      </div>
    )
  }
}

export default Relay.createContainer(Artwork, {
  fragments: {
    artwork: () => Relay.QL`
      fragment on Artwork {
        image {
          placeholder
          url(version: "large")
          aspect_ratio
        }
       ${ArtworkMetadata.getFragment("artwork")}
      }
    `,
  },
})

interface RelayProps {
  artwork: {
    image: {
      placeholder: string | null,
      url: string | null,
      aspect_ratio: number | null,
    } | null,
  },
}
