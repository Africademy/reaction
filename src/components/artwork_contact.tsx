import * as React from "react"
import * as Relay from "react-relay"

export class ArtworkContact extends React.Component<RelayProps, null> {

  contactLine() {
    const artwork = this.props.artwork
    if (artwork.sale && artwork.sale.is_auction) {
      return this.auctionLine()
    } else if (artwork.is_inquireable) {
      return this.contactPartnerLine()
    }
  }

  auctionLine() {
    const artwork = this.props.artwork
    if (artwork.sale.is_live_open) {
      return (
        <a href={artwork.href}>Enter Live Auction</a>
      )
    } else if (artwork.sale.is_open) {
      const sa = artwork.sale_artwork
      const bids = sa.counts.bidder_positions
      if (bids > 0) {
        const s = bids > 1 ? 's' : ''
        return (
          <span>
            {sa.highest_bid.display}(bid{s})
          </span>
        )
      } else {
        return (
          <span>
            {sa.opening_bid.display}
          </span>
        )
      } else if (artwork.sale.is_closed) {
        return (<span>Auction closed</span>)
      } else {
        return (<span/>)
      }
    }
  }

  contactPartnerLine() {
    return (
      <a href={this.props.artwork.href}>
        Contact {this.props.artwork.partner.type.toLowerCase()}
      </a>
    )
  }

  render() {
    return (
      <div>
        {this.contactLine()}
      </div>
    )
  }
}

export default Relay.createContainer(ArtworkContact, {
  fragments: {
    artwork: () => Relay.QL`
      fragment on Artwork {
        _id
        href
        is_inquireable
        sale {
          is_auction
          is_live_open
          is_open
          is_closed
        }
        sale_artwork {
          highest_bid {
            display
          }
          opening_bid {
            display
          }
          counts {
            bidder_positions
          }
        }
        partner(shallow: true) {
          type
          href
        }
      }
    `
  },
})
