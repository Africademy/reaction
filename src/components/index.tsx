import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import Buttons from "./buttons/index"
import Modal from "./modal/modal"

import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Checkbox from "./checkbox"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
