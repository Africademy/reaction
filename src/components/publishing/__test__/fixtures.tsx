import { Props as ImageSetPreviewProps } from "../imageset_preview"

export const Artworks = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
    },
    artist: {
      name: "Fernando Botero",
    },
    width: 1152,
    height: 826,
  },
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
      {
        name: "Frida Kahlo",
        slug: "frida-kahlo",
      },
    ],
    width: 1152,
    height: 826,
  },
]

export const Articles = [
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "classic",
    vertical: null,
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "standard",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "feature",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
]

export const Authors = [
  {
    image_url: "https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg",
    name: "Abigail Cain",
    bio: "[Abigail Cain](https://artsy.net/abigail) is an Art Market Editor at Artsy",
    twitter_handle: "abigailcain",
  },
  {
    image_url: "http://files.artsy.net/images/molly_01.jpg",
    name: "Molly Gottschalk",
    bio:
      "[Molly Gottschalk](https://artsy.net/molly) is a Features Producer at Artsy. She is a photographer and lover of cats.",
    twitter_handle: "mollygottschalk",
  },
  {
    name: "Halley Johnson",
    bio: "[Halley Johnson](https://artsy.net/molly) is an Writer Guru at Artsy. She enjoys speaking Georgian.",
    twitter_handle: "halleyjohnson",
  },
]

export const Embeds = [
  {
    mobile_height: 1300,
    height: 1000,
    url: "http://files.artsy.net/documents/1parrasch.html",
    layout: "overflow",
    type: "embed",
  },
]

export const HeroSections = [
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]

export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini: ImageSetPreviewProps["section"] = {
  type: "image_set",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetFullSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  images: Images,
}

export const ImageSetMiniSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  images: Images,
}
export const Videos = [
  {
    url: "https://www.youtube.com/watch?v=PXi7Kjlsz9A",
    caption: "<p>What motivates patrons to fund artists’ wildest dreams?</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
    caption:
      "<p>2016 was a memorable year for the world, and art along with it. Powered by data culled from Artsy as well as UBS’s Planet Art app, “The Year in Art 2016” will explore how the creative community responded to the cultural shifts and tribulations this year has seen—from the destruction of Palmyra to the proliferation of Virtual Reality to the U.S. election.</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/ditbyaUgdcl6mHin07TfKA%2FMassimilianoGioni_0581.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
  },
]
