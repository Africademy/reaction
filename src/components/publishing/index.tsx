import Article from "./article"
import FeatureHeader from "./header/feature_header"
import Header from "./header/header"
import Artwork from "./sections/artwork"
import Authors from "./sections/authors"
import Caption from "./sections/caption"
import Embed from "./sections/embed"
import FullscreenViewer from "./sections/fullscreen_viewer/fullscreen_viewer"
import Image from "./sections/image"
import ImageCollection from "./sections/image_collection"
import ImageSetPreview from "./sections/imageset_preview"
import ImageSetPreviewClassic from "./sections/imageset_preview_classic"
import Text from "./sections/text"
import Video from "./sections/video"

// Icon SVGs
import ImageSet from "./icon/image_set"
import Remove from "./icon/remove"

export default {
  Article,
  Artwork,
  Authors,
  Embed,
  Caption,
  FeatureHeader,
  FullscreenViewer,
  Header,
  Image,
  ImageCollection,
  ImageSetPreview,
  ImageSetPreviewClassic,
  Text,
  Video,
  Icon: {
    ImageSet,
    Remove,
  },
}
