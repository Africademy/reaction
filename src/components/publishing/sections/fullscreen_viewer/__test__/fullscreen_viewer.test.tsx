import * as React from "react"
import * as renderer from "react-test-renderer"
import { Images } from "../../../fixtures/components"
import FullscreenViewer from "../fullscreen_viewer"

jest.mock("react-slick", () => {
  const React = require("react")
  return props => <div>{props.children}</div>
})

it("renders properly", () => {
  const onClose = jest.fn()
  const viewer = renderer.create(<FullscreenViewer images={Images} show onClose={onClose} />).toJSON()
  expect(viewer).toMatchSnapshot()
})
