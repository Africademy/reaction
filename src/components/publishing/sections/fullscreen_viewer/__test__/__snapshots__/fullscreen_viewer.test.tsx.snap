// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="file__FullscreenViewerContainer-s1adxe1n-0 dpQpeR"
>
  <div>
    <div>
      <div
        className="file__SlideContainer-s12qvrc9-0 dbCrtu"
      >
        <div
          className="file__Title-s12qvrc9-2 yQTKs"
        />
        <div
          className="file__Image-s12qvrc9-1 gIalgO"
          src="https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg"
        />
        <div
          className="file__CaptionContainer-s1koc55t-2 bWybux"
        >
          <div
            className="file__CaptionTextContainer-s1koc55t-3 eXwGWE"
          >
            <div
              className="file__StyledCaptionToggle-s1koc55t-0 jiJlYL"
              onClick={[Function]}
            >
              <span>
                View Caption
              </span>
            </div>
            <div
              className="sc-bxivhb jIhHWs"
              open={false}
            >
              <div
                className="sc-htpNat dqiQen"
              >
                <div
                  className="artist-name file__Line-s7zcrfi-1 jFJBbs"
                >
                  <span
                    className="name"
                  >
                    <a
                      className="file__StyledTextLink-s18afosz-0 hOUyga"
                      href="/artist/fernando-botero"
                    >
                      Fernando Botero
                    </a>
                  </span>
                </div>
                <div
                  className="file__Line-s7zcrfi-1 jFJBbs"
                >
                  <span
                    className="title"
                  >
                    <a
                      className="file__StyledTextLink-s18afosz-0 hOUyga"
                      href="/artwork/fernando-botero-nude-on-the-beach"
                    >
                      Nude on the Beach
                    </a>
                  </span>
                  <span
                    className="comma"
                  >
                    , 
                  </span>
                  <span
                    className="date"
                  >
                    2000
                  </span>
                  <span
                    className="comma"
                  >
                    , 
                  </span>
                  <a
                    className="file__StyledTextLink-s18afosz-0 hOUyga"
                    href="/partner/gary-nader"
                  >
                    Gary Nader
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="file__Index-s1koc55t-1 gCANWd"
          >
            1 of 3
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="file__SlideContainer-s12qvrc9-0 dbCrtu"
      >
        <div
          className="file__Title-s12qvrc9-2 yQTKs"
        />
        <div
          className="file__Image-s12qvrc9-1 hTUeaS"
          src="https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg"
        />
        <div
          className="file__CaptionContainer-s1koc55t-2 bWybux"
        >
          <div
            className="file__CaptionTextContainer-s1koc55t-3 eXwGWE"
          >
            <div
              className="file__StyledCaptionToggle-s1koc55t-0 jiJlYL"
              onClick={[Function]}
            >
              <span>
                View Caption
              </span>
            </div>
            <div
              className="sc-bxivhb jIhHWs"
              open={false}
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Photo by Adam Kuehl for Artsy.</p>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="file__Index-s1koc55t-1 gCANWd"
          >
            2 of 3
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="file__SlideContainer-s12qvrc9-0 dbCrtu"
      >
        <div
          className="file__Title-s12qvrc9-2 yQTKs"
        />
        <div
          className="file__Image-s12qvrc9-1 Knxds"
          src="https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg"
        />
        <div
          className="file__CaptionContainer-s1koc55t-2 bWybux"
        >
          <div
            className="file__CaptionTextContainer-s1koc55t-3 eXwGWE"
          >
            <div
              className="file__StyledCaptionToggle-s1koc55t-0 jiJlYL"
              onClick={[Function]}
            >
              <span>
                View Caption
              </span>
            </div>
            <div
              className="sc-bxivhb jIhHWs"
              open={false}
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
                  }
                }
              />
            </div>
          </div>
          <div
            className="file__Index-s1koc55t-1 gCANWd"
          >
            3 of 3
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="file__Close-s1adxe1n-1 eKSoES"
    onClick={[Function]}
  >
    <div
      className="file-s1ekx40u-0 drQqZd"
      style={undefined}
    >
      î˜”
    </div>
  </div>
</div>
`;
